### PR TITLE
graphqlbackend: fix typed-nil error

### LINKED ISSUE #4671: panic: nil error when calling `ListAll()` on zoekt backend
@tsenart linked me to this panic log on Google Cloud: https://console.cloud.google.com/errors/CNffpKuZvIaCAg?time=P1D&project=sourcegraph-dev&authuser=1

Here is the stack trace:

```
runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe467c2]
at github.com/sourcegraph/sourcegraph/pkg/search/backend.(*Zoekt).Enabled (text.go:76)
at github.com/sourcegraph/sourcegraph/pkg/search/backend.(*Zoekt).ListAll (text.go:45)
at github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.zoektIndexedRepos (textsearch.go:835)
at github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.searchFilesInRepos (textsearch.go:894)
at github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).suggestFilePaths (search.go:576)
at github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).Suggestions.func2 (search_suggestions.go:89)
at github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend.(*searchResolver).Suggestions.func5 (search_suggestions.go:191)
```

This is probably related to this commit I made earlier today in which the zoekt backend is injected as a dependency: https://github.com/sourcegraph/sourcegraph/commit/26df02becae2bc4911fede12556a5a24320bea51

The curious bit is that the there's a nil check right at the top of the function:

https://github.com/sourcegraph/sourcegraph/blob/26df02becae2bc4911fede12556a5a24320bea51/cmd/frontend/graphqlbackend/textsearch.go#L813-L815

and that it fails here after `ListAll` is called anyway:

https://github.com/sourcegraph/sourcegraph/blob/26df02becae2bc4911fede12556a5a24320bea51/pkg/search/backend/text.go#L76

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes #4671

Passing in `nil` into `zoektIndexedRepos` caused the infamous "typed
nil" error in Go to pop up, which ultimately caused `ListAll()` to be
called in `zoektIndexedRepos`, which caused a nil-dereference panic.

Test plan: go test

### CODE REVIEW

**Comment by mrnugget** [comment]
This is basically how the panic is produced:

```go
func Test_zoektIndexedRepos_typedNil(t *testing.T) {
	repos := makeRepositoryRevisions("foo/indexed-one@")
	z := IndexedSearch()
	z = nil
	_, _, _, _ = zoektIndexedRepos(context.Background(), z, repos)
}
```

Full explanation of typed-nil [can be found here](https://dev.to/pauljlucas/go-tcha-when-nil--nil-hic)

File: cmd/frontend/graphqlbackend/textsearch.go [hunk 1]
```diff
@@ -887,17 +884,21 @@ func searchFilesInRepos(ctx context.Context, args *search.Args) (res []*fileMatc
...
-		tr.LogFields(otlog.String("indexErr", err.Error()))
-		log15.Warn("zoektIndexedRepos failed", "error", err)
-		common.indexUnavailable = true
-		err = nil
+	var (
+		zoektRepos       []*search.RepositoryRevisions
+		searcherRepos    []*search.RepositoryRevisions = args.Repos
+		indexedRevisions map[*search.RepositoryRevisions]string
+	)
+	if IndexedSearch().Enabled() {
```
Comment (attfarhan) [inline cmd/frontend/graphqlbackend/textsearch.go]: If indexed search is not enabled, which is the case on Sourcegraph.com, will `searcherRepos` always be an empty list now? Currently, when all the error occurs, all repos are returned as `searcherRepos`.

Comment (mrnugget) [reply cmd/frontend/graphqlbackend/textsearch.go]: It's the same behavior as before I changed the interface of the function, see here: https://github.com/sourcegraph/sourcegraph/commit/26df02becae2bc4911fede12556a5a24320bea51

Comment (attfarhan) [reply cmd/frontend/graphqlbackend/textsearch.go]: Ah ok got it sorry was looking at the wrong revision.

**Comment by keegancsmith** [comment]
thanks for the quick fix!

**Pull request merged**